common/unused_port/src/lib.rs

```diff
@@ -6,14 +6,30 @@ pub enum Transport {
     Udp,
 }
 
-/// A convenience function for `unused_port(Transport::Tcp)`.
-pub fn unused_tcp_port() -> Result<u16, String> {
-    unused_port(Transport::Tcp)
+#[derive(Copy, Clone)]
+pub enum IpVersion {
+    Ipv4,
+    Ipv6,
+}
+
+/// A convenience warpper over [`zero_port`].
+pub fn unused_tcp4_port() -> Result<u16, String> {
+    zero_port(Transport::Tcp, IpVersion::Ipv4)
+}
+
+/// A convenience warpper over [`zero_port`].
+pub fn unused_udp4_port() -> Result<u16, String> {
+    zero_port(Transport::Udp, IpVersion::Ipv4)
+}
+
+/// A convenience warpper over [`zero_port`].
+pub fn unused_tcp6_port() -> Result<u16, String> {
+    zero_port(Transport::Tcp, IpVersion::Ipv6)
 }
 
-/// A convenience function for `unused_port(Transport::Tcp)`.
-pub fn unused_udp_port() -> Result<u16, String> {
-    unused_port(Transport::Udp)
+/// A convenience warpper over [`zero_port`].
```

Comment:
```suggestion
/// A convenience wrapper over [`zero_port`].
```
Contributing :sunglasses: